### PR TITLE
Core widgets: new signals

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Cannot connect core Orange widgets to text add-on widgets which accept corpus.

##### Description of changes
Something similar as in #2312 .

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
Perhaps a little checklist: 😊 
- [X] Select Columns
- [X] Select Rows
- [X] Data Sampler
- [X] Table
- [X] Randomize
- [X] Edit Domain
- [ ] Confusion Matrix
- [ ] Hierarchical Clustering
- [ ] Scatter Plot
- [ ] MDS
- [ ] Linear Projection
- [ ] Box Plot
- [ ] Sieve Diagram
- [ ] Heat Map
- [ ] Tree
- [ ] k-Means
- [ ] Distance Map
- [ ] Distance Matrix

Edit if I missed something. The list is tentatively ordered by priority.

File: Orange/widgets/data/owtable.py [hunk 1]
```diff
@@ -367,9 +368,12 @@ class OWDataTable(widget.OWWidget):
...
 
-    inputs = [("Data", Table, "set_dataset", widget.Multiple)]
-    outputs = [("Selected Data", Table, widget.Default),
-               (ANNOTATED_DATA_SIGNAL_NAME, Table)]
+    class Inputs:
+        data = Input("Data", Table, multiple=True)
+
+    class Outputs:
+        selected_data = Output("Selected Data", Table, default=True)
+        adsn = Output(ANNOTATED_DATA_SIGNAL_NAME, Table)
```
Comment (nikicc) [inline Orange/widgets/data/owtable.py]: Why not just `annotated_data`?

File: Orange/widgets/visualize/owscatterplot.py [hunk 2]
```diff
@@ -109,13 +109,15 @@ class OWScatterPlot(OWWidget):
...
+        data = Input("Data", Table, default=True)
+        data_subset = Input("Data Subset", Table)
+        features = Input("Features", AttributeList)
 
-    outputs = [("Selected Data", Table, Default),
-               (ANNOTATED_DATA_SIGNAL_NAME, Table),
-               ("Features", Table)]
+    class Outputs:
+        selected_data = Output("Selected Data", Table, default=True)
+        adsn = Output(ANNOTATED_DATA_SIGNAL_NAME, Table)
```
Comment (nikicc) [inline Orange/widgets/visualize/owscatterplot.py]: Same here. I would probably use `annotated_data`.

File: Orange/widgets/evaluate/owconfusionmatrix.py [hunk 3]
```diff
@@ -89,9 +89,12 @@ class OWConfusionMatrix(widget.OWWidget):
...
 
-    inputs = [("Evaluation Results", Orange.evaluation.Results, "set_results")]
-    outputs = [("Selected Data", Orange.data.Table, widget.Default),
-               (ANNOTATED_DATA_SIGNAL_NAME, Orange.data.Table)]
+    class Inputs:
+        evaluation_results = Input("Evaluation Results", Orange.evaluation.Results)
+
+    class Outputs:
+        selected_data = Output("Selected Data", Orange.data.Table, default=True)
+        adsn = Output(ANNOTATED_DATA_SIGNAL_NAME, Orange.data.Table)
```
Comment (nikicc) [inline Orange/widgets/evaluate/owconfusionmatrix.py]: Same here 🙈

Comment (jerneju) [inline Orange/widgets/evaluate/owconfusionmatrix.py]: Yes. Totally 13 times. I shall change everything!

Comment (jerneju) [inline Orange/widgets/evaluate/owconfusionmatrix.py]: Plus 28 times in tests.

Comment (kernc) [inline Orange/widgets/evaluate/owconfusionmatrix.py]: And perhaps in general don't use non-standard abbreviations, acronyms in particular!

**Pull request merged**